### ISSUE TITLE
Confusing attribute names in PairwiseAligner

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#1992]
The match and mismatch scores are stored in the `match` and `mismatch` attributes in `PairwiseAligner`. The gap scores explicitly have the word `score` in the attribute name, for example `target_open_gap_score`. A user may therefore expect the match and mismatch scores to be named `match_score` and `mismatch_score`. To make matters worse, printing the aligner suggests that these attribute names are indeed `match_score` and `mismatch_score`, for example:

```python
>>> print aligner
Pairwise sequence aligner with parameters
  match score: 1.000000
  mismatch score: -4.000000
  target open gap score: -4.000000
  target extend gap score: -1.000000
...
```

We should either change `__str__` to be consistent with the actual attribute names, or change the attribute names to `match_score` and `mismatch_score`. It may be better also to raise an Exception if a user tries to create a new attribute; this would also prevent confusion for the correct name of attributes (e.g. `target_gap_open_score` instead of `target_open_gap_score`).

**Comment by peterjc** [comment]
Changing the ``__str__`` output seems less likely to cause users trouble, while renaming would break backward compatibility. You could add the alternative names as alias properties?

**Comment by mdehoon** [comment]
Fixed by #2001
